app/api/v1/clan_icons.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from fastapi import APIRouter
+from fastapi import Depends
+from fastapi import File
+from fastapi import Response
+
+import app.usecases.images
+import settings
+from app.adapters import s3
+from app.api.authorization import AdminAuthorization
+from app.api.authorization import authorize_admin
+from app.errors import Error
+from app.errors import ErrorCode
+from app.usecases.images import ImageType
+
+router = APIRouter()
+
+
+def _get_status_code_for_error(error_code: ErrorCode) -> int:
+    try:
+        return {
+            ErrorCode.INVALID_CONTENT: 400,
+            ErrorCode.INAPPROPRIATE_CONTENT: 400,
+            ErrorCode.SERVICE_UNAVAILABLE: 503,
+        }[error_code]
+    except KeyError:
+        logging.warning(
+            "Unmapped error code while resolving http status code",
+            extra={"error_code": error_code},
+        )
+        return 500
+
+
+@router.post("/api/v1/clans/{clan_id}/icon")
+async def upload_avatar(
+    clan_id: str,
```

Comment:
should this be an int?

Reply:
I think the avatar one should also be an int

Reply:
should probably be fixed on avatars as well?